### ISSUE TITLE
Possible false positive on duplicate import rule

### ISSUE THREAD

**Issue by doctorcolinsmith** [realm/SwiftLint#2594]
### New Issue Checklist

- [X] Updated SwiftLint to the latest version
- [X] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

I think there may be a possible false positive for the duplicate import rule.  The following snippet triggers the warning "Duplicate Imports Violation: Imports should be unique. (duplicate_imports)"

```
import SceneKit
import SceneKit.ModelIO
```

By default SceneKit does not import all of the framework features.  To get the model I/O functions, the ModelIO submodule also needs to be imported.

This was tested with SwiftLint 0.30.1, in 10.1, building an iOS app.

**Comment by jpsim** [comment]
Yes this is a false positive, and one we considered when writing the rule. Most of Apple’s modules with sub modules like these are re-exported from the parent (e.g. `Foundation.NSString`), so we decided to have the rule assume this more common case and require silencing these false positives.

That being said I think we could whitelist this case and further allow the rule to be configured with a list of required explicit submodule imports to handle the 3rd party exceptions too (e.g. `Realm.Private` & `Realm.Dynamic`).

**Comment by doctorcolinsmith** [comment]
I like the idea of creating whitelist for the rule.  It would be easy to add such modules as an exception to a .swiftlint.yml file.

**Comment by nevil** [comment]
I guess the below could also be whitelisted:
```swift
import UIKit
import UIKit.UIGestureRecognizerSubclass
```

**Comment by harlanhaskins** [comment]
This rule also trips for scoped imports, which are used to resolve ambiguities when importing.

i.e.

```swift
import FirstModule // defines type FirstType
import SecondModule // also defines type FirstType

import struct SecondModule.FirstType // resolves the ambiguity and allows for unqualified references to `FirstType` to refer to `SecondModule.FirstType`.
```
